src/mon/MgrMonitor.cc

```diff
@@ -1012,13 +1012,24 @@ bool MgrMonitor::preprocess_command(MonOpRequestRef op)
       f->dump_object("mgrmap", m);
     }
     f->flush(rdata);
-  } else if (prefix == "mgr module ls") {
+} else if (prefix == "mgr module ls") {
     if (f) {
+      // Create a mapping from module names to ModuleInfo pointers
+      std::unordered_map<std::string, const MgrMap::ModuleInfo*> module_info_map;
+      const auto always_on_modules = map.get_always_on_modules();
+      for (const auto& mi : map.available_modules) {
+        module_info_map[mi.name] = &mi;
+      }
+
       f->open_object_section("modules");
       {
+        // always_on_modules
```

Comment:
> If you need a block - better just name it (as an aux function)

Would it be okay if I just remove the block, instead of creating a whole new function?